### PR TITLE
Support exclusion file for Qute for validation

### PR DESCRIPTION
**Pull request by angelozerr**
Support exclusion file for Qute for validation

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
This PR requires https://github.com/redhat-developer/vscode-quarkus/pull/439

For the moment code action stores the full template uri as pattern:

QuteExcludeFileValidation

My idea is not to store the full path but a pattern with a given file name. For this sample, I think code action should store 

`"**/items.qute.*"`  

@fbricon @mkouba what do you think about that?

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteCodeActions.java [hunk 1]
```diff
@@ -59,6 +59,10 @@
 
 	private static final String DISABLE_VALIDATION_TITLE = "Disable Qute validation.";
 
+	private static final String QUTE_VALIDATION_EXCLUDED_SECTION = "quarkus.tools.qute.validation.excluded";
+
+	private static final String EXCLUDED_VALIDATION_TITLE = "Exclude this file for validation.";
```
Comment (fbricon) [inline qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteCodeActions.java:64]: Exclude this file from validation.

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteCodeActions.java [hunk 2]
```diff
@@ -140,10 +144,18 @@ private void doCodeActionsForUndefinedVariable(Template template, Diagnostic dia
...
 
-	public void doCodeActionToDisableValidation(List<Diagnostic> diagnostics, List<CodeAction> codeActions) {
+	public void doCodeActionToDisableValidation(Template template, List<Diagnostic> diagnostics, List<CodeAction> codeActions) {
+		// Disable Qute validation
 		CodeAction disableValidationQuickFix = createConfigurationUpdateCodeAction(DISABLE_VALIDATION_TITLE,
 				QUTE_VALIDATION_ENABLED_SECTION, false, ConfigurationItemEditType.update, diagnostics);
 		codeActions.add(disableValidationQuickFix);
+
+		// Disable Qute validation for the given template
+		String pattern = template.getUri();
```
Comment (fbricon) [inline qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteCodeActions.java:154]: remove workspace root from pattern

**Comment by angelozerr** [comment]
Close for https://github.com/redhat-developer/quarkus-ls/pull/544

**Pull request closed without merging**